generator/shared.go

```diff
@@ -113,6 +166,21 @@ func GoLangOpts() *LanguageOpts {
 		opts.Comments = true
 		return imports.Process(ffn, content, opts)
 	}
+	opts.fileNameFunc = func(name string) string {
+		// whenever a generated file name ends with a suffix
+		// that is meaningful to go build, adds a "swagger"
+		// suffix
+		parts := strings.Split(swag.ToFileName(name), "_")
+		_, isArch := goarchValues[parts[len(parts)-1]]
+		_, isOs := goosValues[parts[len(parts)-1]]
+		_, isOtherReserved := goOtherReservedSuffixes[parts[len(parts)-1]]
+		if isArch || isOs || isOtherReserved {
```

Comment:
This checks all three maps even when the first one matches.  It would be better to short circuit as soon as one matches.  For that matter, there is no reason that you need 3 maps here.  A single map with all reserved suffixes should be fine.  Then you only need to access the last part once and can do this:

    if goReservedSuffixes[len(parts)-1] {
    ...

Reply:
ok done. Thanks